lib/ast.js

```diff
@@ -589,61 +589,64 @@ var AST_Seq = DEFNODE("Seq", "car cdr", {
         cdr: "[AST_Node] second element in sequence"
     },
     $cons: function(x, y) {
+        if (!x) {
+            throw new Error('AST_Seq.car missing');
+        }
+        if (!y) {
+            throw new Error('AST_Seq.cdr missing');
+        }
+        if (x instanceof AST_Seq) {
+            throw new Error('AST_Seq.car cannot be AST_Seq');
+        }
         var seq = new AST_Seq(x);
         seq.car = x;
         seq.cdr = y;
         return seq;
     },
     $from_array: function(array) {
         if (array.length == 0) return null;
-        if (array.length == 1) return array[0].clone();
```

Comment:
When I see a clone() being dropped there is the concern that the user of this function wants to operate on a copy of the data without affecting the original.